pkg/southbound/synchronizer/factory.go

```diff
@@ -70,7 +72,40 @@ func Factory(topoChannel <-chan *topodevice.ListResponse, opStateChan chan<- eve
 				}
 				synchronizers[device.ID] = synchronizer
 				go synchronizer.connect()
+			} else if ok && topoEvent.Type == topodevice.ListResponse_UPDATED {
+				changed := false
+				if synchronizer.device.Address != topoEvent.Device.Address {
+					oldAddress := synchronizer.device.Address
+					synchronizer.mu.Lock()
+					synchronizer.device.Address = topoEvent.Device.Address
+					changed = true
+					synchronizer.mu.Unlock()
+					log.Infof("Topo device %s UPDATED address %s -> %s ", device.ID, oldAddress, topoEvent.Device.Address)
+					synchronizer.close()
+					time.Sleep(maxBackoffTime + time.Millisecond*10) // close might not take effect until timeout
```

Comment:
is there a way to avoid the sleep here, maybe making the close() method return something and thus making it synchronous ?

Reply:
The reality is that it should block for the longer of either
* the gRPC timeout OR
* the retry value (up to maxBackoffTime)

The reason why that there is currently no way of interrupting Blocking a DialContext() call - as is made through the gNMI client. This means we have to wait until the DialContext() times out in case we are dealing with a non-reachable device.

See the issue #981